docs/data/sdks/typescript-node/migration.md

```diff
@@ -3,6 +3,158 @@ title: Node.JS SDK Migration Guide
+
+## Instrumentation
+
+New Node.js SDK offers an API to instrument events. To migrate to it, you need to update a few calls. The following sections detail which calls have changed.
+
+### Initialization
+
+The maintenance Node.js SDK only supports namespace import. The new Node.js SDK supports namespace import (`import * as amplitude from '@amplitude/analytics-node'`) and named import (`import { init } from '@amplitude/analytics-node'`) as well. We are using named import in the examples of new Node.js SDK in this documentation.
+
+To initialize the SDK, call `init()`, with the user ID and configuration parameters.
+
+```diff
+- import * as Amplitude from '@amplitude/node'
++ import { init } from '@amplitude/analytics-node';
+
+var options = {};
+- const client = Amplitude.init(AMPLITUDE_API_KEY, options);
++ init(API_KEY, options);
+```
+
+### Configuration
+
+The new Node.js SDK configuration comes in a different shape. Some configurations are no longer supported.
+
+|@amplitude/node|@amplitude/analytics-node|
+|-|-|
+| `debug` | `logLevel` set to WARN level|
+| `logLevel` | `logLevel` |
+| `maxCachedEvents` | `flushQueueSize` |
+| `retryTimeouts` | `flushMaxRetries` can only be set to a number instead of an array of number as in `retryTimeouts`
+| `optOut` | `optOut` |
+| `retryClass` | CUSTOMIZATION NOT SUPPORT. Retry logic is handled by new Node.js SDK|
+| `transportClass` | `transportProvider` |
+| `serverUrl` | `serverUrl` |
+| `uploadIntervalInSec` | `flushIntervalMillis` is in milliseconds while `uploadIntervalInSec` is in seconds|
+| `minIdLength` | `minIdLength` |
+| `requestTimeoutMillis` | NOT SUPPORT |
+| `onRetry` | CUSTOMIZATION NOT SUPPORT. Retry logic is handled by the new Node.js SDK |
+
+### Tracking events
+
+#### `logEvent()`
+
+The `logEvent()` API maps to `track()`.
+
+```diff
++ import { track } from '@amplitude/analytics-node';
+
+const eventProperties = {
+    buttonColor: 'primary',
+};
+
+- client.logEvent({
+-     event_type: 'Button Clicked',
+-     user_id: 'user@amplitude.com',
+-     event_properties: eventProperties
+- });
++ track('Button Clicked', eventProperties, {
++     user_id: 'user@amplitude.com',
++ });
+```
+
+#### `flush()`
+
+The `flush()` API remains the same.
+
+```diff
++ import { flush } from '@amplitude/analytics-node';
+
+- client.flush();
++ flush();
+```
+
+### Set user properties
+
+#### `identify()`
+
+The `identify()` API is very similar but has a different signature. The [old Node.js SDK](https://github.com/amplitude/Amplitude-Node/blob/2ef295e1fb698286d606ea4a2ccbbfdc4ba3fdc8/packages/node/src/nodeClient.ts#L142) has a signature `(userId: string | null, deviceId: string | null, identify: Identify)` while the [new Node.js SDK](https://github.com/amplitude/Amplitude-TypeScript/blob/8f4ea010279fb21190a2c0595d4ae8a7d9e987ce/packages/analytics-core/src/core-client.ts#L62) has a signature `(identify: Identify, eventOptions?: EventOptions)`.
```

Comment:
Can you use maintenance, and latest instead of old and new? Please align the word in other places.

Reply:
Do you think we should mention the deiceId and the userId in eventOptions can overwrite set the client deviceId and client userId? and what eventOptions look like.

Reply:
Replace all old, new -> maintenance, latest to align with other docs. 

It makes sense to add what `eventOptions` is like. How about [a link to github](https://github.com/amplitude/Amplitude-TypeScript/blob/8f4ea010279fb21190a2c0595d4ae8a7d9e987ce/packages/analytics-types/src/base-event.ts#L12) Feel like it would be enough

> Do you think we should mention the deiceId and the userId in eventOptions can overwrite set the client deviceId and client userId?
Could you elaborate on this?

Reply:
We had a quick call and agree that
- use github page link
- only next-gen client-side `identify()` also sets user id and device id if they are in event options, but node.js is a server-side SDK